### PR TITLE
repl() to open repl

### LINKED ISSUE #127: ✋repl() to pause and open REPL
Problem: Make it easy to fix a part of a test.

Solution: 

- `await repl()` that will open a REPL
- `resume()` to close REPL and continue the rest of the file

### PR DESCRIPTION
**Pull request by jperl**
Resolves #127

To Do:

- [x] on repl close, call resume
- [x] test it works on `node script.js`
- [x] setup basic reporter
- [x] `npx qawolf test --repl` that sets up reporter
- [x] make `repl` a global
- [x] add tests

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -53,6 +53,10 @@ program
   .option("-p, --path <path>", "path to test code")
   .option("--all-browsers", "run tests on chromium, firefox, and webkit")
   .option("--chromium", "run tests on chromium")
+  .option(
+    "--repl",
+    "override the default Jest reporter since it breaks the repl"
```
Comment (flaurida) [inline packages/cli/src/index.ts]: this doesn't seem like what the command really does? something more like "use interactive repl to run code?"

Comment (flaurida) [reply packages/cli/src/index.ts]: (does from a user perspective)

File: packages/repl/src/repl.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { bold } from "kleur";
+import { start, REPLServer } from "repl";
+const { addAwaitOutsideToReplServer } = require("await-outside");
+
+export const repl = (
+  context: any = {},
+  onReplCreated?: (replServer: REPLServer) => void
+) => {
+  logger.debug("repl: start");
+  console.log(bold().yellow("Type .exit or resume() to close the repl"));
```
Comment (flaurida) [inline packages/repl/src/repl.ts]: do `.exit` and `resume()` do the same thing? I doubt it...let's clarify this message

File: packages/repl/src/repl.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { logger } from "@qawolf/logger";
+import { bold } from "kleur";
+import { start, REPLServer } from "repl";
+const { addAwaitOutsideToReplServer } = require("await-outside");
```
Comment (flaurida) [inline packages/repl/src/repl.ts]: import doesn't work here? just confirming

File: packages/repl/src/repl.ts (hunk 2 above)
Comment (jperl) [reply packages/repl/src/repl.ts]: Good point. Removing resume()

File: packages/repl/src/repl.ts (hunk 3 above)
Comment (jperl) [reply packages/repl/src/repl.ts]: They do not have typescript support, so I am just leaving as is. Will add a comment.

Comment (jperl) [reply packages/repl/src/repl.ts]: I'll add types 👍

Comment (jperl) [reply packages/repl/src/repl.ts]: I added types

**Pull request merged**